e2e/api/testingSupport.js

```diff
@@ -31,5 +31,28 @@ module.exports =  {
     }, MAX_RETRIES, RETRY_TIMEOUT_MS);
     if (incidentMessage)
       throw new Error(`Business process failed for case: ${caseId}, incident message: ${incidentMessage}`);
+  },
+  assignCaseToDefendant: async caseId => {
+    const authToken = await idamHelper.accessToken(config.defendantSolicitorUser);
+
+    await retry(() => {
+      return restHelper.request(
+        `${config.url.unspecService}/testing-support/assignCase/${caseId}`,
+        {
+          'Content-Type': 'application/json',
+          'Authorization': `Bearer ${authToken}` },
+        {},
+        'POST')
+        .then( response =>  response._status)
+        .then(status => {
+          if (status === 200) {
+            console.log( 'Role created successfully');
+          } else if (status === 409) {
+            console.log('Role already exists!');
+          } else  {
+            throw new Error(`Error occurred with status : ${status}`);
+          }
+        });
+    }, MAX_RETRIES, RETRY_TIMEOUT_MS);
```

Comment:
This method will be retried 300 times every second if an error occurs. If you don't provide these parameters it will try 3 times with 5 seconds wait between.